Source/iop/Iop_McServ.cpp

```diff
@@ -659,6 +663,33 @@ void CMcServ::GetDir(uint32* args, uint32 argsSize, uint32* ret, uint32 retSize,
 	ret[0] = result;
 }
 
+void CMcServ::SetFileInfo(uint32* args, uint32 argsSize, uint32* ret, uint32 retSize, uint8* ram)
+{
+	auto cmd = reinterpret_cast<const CMD*>(args);
+	CLog::GetInstance().Print(LOG_NAME, "SetFileInfo(port = %d, slot = %d, name = '%s');\r\n", cmd->port, cmd->slot, cmd->name);

```

Comment:
There's a `flags` parameter to that function (as shown here: https://github.com/ps2dev/ps2sdk/blob/11500fe1dcec03592e71406cbd9d34b39aa52589/iop/memorycard/mcserv/src/mcserv.c#L1217). Could you also get that value and print it out?

Also, I think it's used to specify what kind of change can be made to the entry, I think it would be interesting to add a warning if it's not something that we expect. Like, here, I think the the only flag we handle in that implementation would be `sceMcFileAttrFile` (ex.: https://github.com/ps2dev/ps2sdk/blob/11500fe1dcec03592e71406cbd9d34b39aa52589/iop/memorycard/mcman/src/ps2mc_fio.c#L485).

Reply:
Done. I hope the enum is okay this way? Maybe we can fold it together with `OPEN_FLAGS`, but I would suppose that should be done in a separate step afterwards.

Reply:
Yup, all good!